File: U1/EXAMEN_SOAP_DOTNET_GR01/01 SERVIDOR/BanquitoServer_Soap_DotNet_GR01/BanquitoServer_Soap_DotNet_GR01/Models/Credito.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace BanquitoServer_Soap_DotNet_GR01.Models
{
    /// <summary>
    /// Entidad Crédito
    /// </summary>
    [Table("Credito")]
    public class Credito
    {
        [Key]
        [Column("CreditoId")]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public long CreditoId { get; set; }

        [Required]
        [Column("NumeroCredito")]
        [MaxLength(20)]
        [Index("IX_Credito_NumeroCredito", IsUnique = true)]
        public string NumeroCredito { get; set; }

        [Required]
        [Column("MontoCredito")]
        public decimal MontoCredito { get; set; }

        [Required]
        [Column("TasaInteres")]
        public decimal TasaInteres { get; set; }

        [Required]
        [Column("NumeroCuotas")]
        public int NumeroCuotas { get; set; }

        [Required]
        [Column("CuotaMensual")]
        public decimal CuotaMensual { get; set; }

        [Required]
        [Column("FechaOtorgamiento")]
        public DateTime FechaOtorgamiento { get; set; }

        [Required]
        [Column("Estado")]
        [MaxLength(20)]
        [Index("IX_Credito_Estado")]
        public string Estado { get; set; }

        [Column("Descripcion")]
        [MaxLength(200)]
        public string Descripcion { get; set; }

        // Foreign Key
        [Required]
        [Column("ClienteId")]
        [ForeignKey("Cliente")]
        [Index("IX_Credito_ClienteId")]
        public long ClienteId { get; set; }

        // Propiedades de navegación
        public virtual Cliente Cliente { get; set; }
        public virtual ICollection<CuotaAmortizacion> CuotasAmortizacion { get; set; }

        public Credito()
        {
            CuotasAmortizacion = new HashSet<CuotaAmortizacion>();
            Estado = "ACTIVO";
            FechaOtorgamiento = DateTime.Now;
        }
    }
}

File: U2/PAO202551_ELECTRO_BDDNUBE_RESTFULL_JAVA_GR01/04. SERVIDOR/BanquitoServer_Restfull_Java_GR01/src/main/java/ec/edu/monster/Models/Credito.java
package ec.edu.monster.Models;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

/**
 * Entidad Crédito
 */
@Entity
@Table(name = "Credito")
@Data
@NoArgsConstructor
public class Credito {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "CreditoId")
    private Long creditoId;

    @Column(name = "NumeroCredito", nullable = false, length = 20, unique = true)
    private String numeroCredito;

    @Column(name = "MontoCredito", nullable = false, precision = 12, scale = 2)
    private BigDecimal montoCredito;

    @Column(name = "TasaInteres", nullable = false, precision = 5, scale = 4)
    private BigDecimal tasaInteres;

    @Column(name = "NumeroCuotas", nullable = false)
    private Integer numeroCuotas;

    @Column(name = "CuotaMensual", nullable = false, precision = 12, scale = 2)
    private BigDecimal cuotaMensual;

    @Column(name = "FechaOtorgamiento", nullable = false)
    private LocalDateTime fechaOtorgamiento;

    @Column(name = "Estado", nullable = false, length = 20)
    private String estado = "ACTIVO";

    @Column(name = "Descripcion", length = 200)
    private String descripcion;

    // Foreign Key
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "ClienteId", nullable = false)
    private Cliente cliente;

    // Relación con cuotas de amortización
    @OneToMany(mappedBy = "credito", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    private List<CuotaAmortizacion> cuotasAmortizacion = new ArrayList<>();

    @PrePersist
    public void prePersist() {
        if (fechaOtorgamiento == null) {
            fechaOtorgamiento = LocalDateTime.now();
        }
        if (estado == null) {
            estado = "ACTIVO";
        }
    }
}
